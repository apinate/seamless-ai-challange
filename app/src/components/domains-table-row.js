import React from 'react';
import PropTypes from 'prop-types';

const DomainsTableRow = ({ data: { name, domain } }) => (
  <tr>
    <td >{name}</td>
    <td>{domain}</td>
  </tr>
);

DomainsTableRow.propTypes = {
  data: PropTypes.shape({
    name: PropTypes.string.isRequired,
    domain: PropTypes.string.isRequired,
  }).isRequired,
};

export default DomainsTableRow;
