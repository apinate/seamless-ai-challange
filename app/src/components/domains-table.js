import React from 'react';
import PropTypes from 'prop-types';

import DomainsTableRow from './domains-table-row';

const DomainsTable = (props) => {
  const renderRow = data => <DomainsTableRow key={data.name} data={data} />;

  return (
    <div className="domains-table">
      {props.companies.map(renderRow)}
    </div>
  );
};

DomainsTable.propTypes = {
  companies: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    domain: PropTypes.string.isRequired,
  }).isRequired).isRequired,
};

export default DomainsTable;
