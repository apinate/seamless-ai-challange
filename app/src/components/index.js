import DomainsTable from './domains-table';
import SearchBar from './search-bar';

export { DomainsTable, SearchBar };
