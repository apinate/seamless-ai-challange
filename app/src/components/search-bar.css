.search-icon-addon {
  background: transparent;
}

.search-input-container {
  margin: 0;
  padding: 0;
}
