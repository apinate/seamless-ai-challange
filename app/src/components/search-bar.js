import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './search-bar.css';

class SearchBar extends Component {
  state = { value: '' };

  onChange = ({ target: { value } }) => {
    this.setState({ value });
  }

  onClick = () => {
    const { queryExtractor, onSearch } = this.props;
    const queryData = (queryExtractor && queryExtractor(this.state.value)) || this.state.value;

    if (this.validate(queryData)) {
      onSearch(queryData);
    }
  }

  get hasError() {
    return this.state.error;
  }

  validate(queryData) {
    let error = '';
    try {
      const { validation } = this.props;
      if (validation) {
        validation(queryData);
      }
    } catch (err) {
      error = err.message;
    }
    this.setState({ error });

    return !error;
  }

  classNamesOnError(classNames) {
    return this.hasError ? classNames : '';
  }

  render() {
    return (
      <div className="mt-3">
        <div className="d-flex flex-wrap flex-row justify-content-between align-content-center">
          <div className={`form-group search-input-container col-lg-10 ${this.classNamesOnError('has-danger')}`}>
            <div className="input-group search-input-group">
              <span className="input-group-addon search-icon-addon">
                <i className="fa fa-search" aria-hidden="true" />
              </span>
              <input
                placeholder={this.props.searchPlaceholder}
                id="search-input"
                type="search"
                value={this.state.value}
                onChange={this.onChange}
                className="form-control"
              />
            </div>
          </div>

          <button className="btn btn-primary" onClick={this.onClick}>Find Domains</button>
        </div>
        <div className={`text-danger ${this.hasError ? '' : 'invisible'}`} >
          <small>{this.state.error || 'No Error'}</small>
        </div>
      </div>
    );
  }
}

SearchBar.propTypes = {
  searchPlaceholder: PropTypes.string.isRequired,
  onSearch: PropTypes.func.isRequired,
  validation: PropTypes.func,
  queryExtractor: PropTypes.func,
};

SearchBar.defaultProps = {
  validation: _ => null,
  queryExtractor: value => value,
};

export default SearchBar;
