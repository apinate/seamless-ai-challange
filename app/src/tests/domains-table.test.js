import React from 'react';
import renderer from 'react-test-renderer';

import { DomainsTable } from '../components';

test('DomainsTable renders correctly when no companies provided', () => {
  const component = renderer.create(<DomainsTable companies={[]} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('DomainsTable renders correctly when one company provided', () => {
  const companies = [
    { name: 'adidas', domain: 'adidas.com' },
  ];

  const component = renderer.create(<DomainsTable companies={companies} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('DomainsTable renders correctly when several companies provided', () => {
  const companies = [
    { name: 'adidas', domain: 'adidas.com' },
    { name: 'nike', domain: 'nike.com' },
    { name: 'apple', domain: 'apple.com' },
  ];

  const component = renderer.create(<DomainsTable companies={companies} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
