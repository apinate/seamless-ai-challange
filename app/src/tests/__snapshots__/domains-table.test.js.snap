// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainsTable renders correctly when no companies provided 1`] = `null`;

exports[`DomainsTable renders correctly when one company provided 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Company Name
      </th>
      <th>
        Domain Name
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        adidas
      </td>
      <td>
        adidas.com
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`DomainsTable renders correctly when several companies provided 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Company Name
      </th>
      <th>
        Domain Name
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        adidas
      </td>
      <td>
        adidas.com
      </td>
    </tr>
    <tr>
      <td>
        nike
      </td>
      <td>
        nike.com
      </td>
    </tr>
    <tr>
      <td>
        apple
      </td>
      <td>
        apple.com
      </td>
    </tr>
  </tbody>
</table>
`;
